src/mds/MDCache.cc

```diff
@@ -2166,7 +2166,8 @@ void MDCache::predirty_journal_parents(MutationRef mut, EMetaBlob *blob,
     // rstat
     if (!primary_dn) {
       // don't update parent this pass
-    } else if (!linkunlink && !(pin->nestlock.can_wrlock(-1) &&
+    } else if (pin->is_auth() && !linkunlink && 
+				!(pin->nestlock.can_wrlock(-1) &&
 				pin->versionlock.can_wrlock())) {
```

Comment:
this change does not seem correct.  CInode::mark_dirty_rstat() calls Locker::mark_updated_scatterlock(), which adds parent inode to updated_scatterlocks list. Later Locker::scatter_tick() triggers scatter-gather processes, which should clear the dirtyrstat flag.